pkg/playground/react-patterns.js

```diff
@@ -40,33 +40,41 @@ import { showMultiTypeaheadDemo } from "./react-demo-multi-typeahead.jsx";
 
 let lastAction = "";
 
-const onDialogStandardClicked = function(mode) {
+const onDialogStandardClicked = function(mode, progress_cb) {
     lastAction = mode;
-    const dfd = cockpit.defer();
-    dfd.notify("Starting something long");
+    let myResolve, myReject;
+    const promise = new Promise((resolve, reject) => {
+        myResolve = resolve;
+        myReject = reject;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers ? It is a bit borderline new (baseline 2024), so if you have doubts, I'm fine with knitting this manually for now.